### PR TITLE
rdb: add wait function to create instance and as a verb

### PR DESCRIPTION
**Pull request by remyleone**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request.
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/scaleway/scaleway-cli/blob/v2/CHANGELOG.md):

```release-note

```

### CODE REVIEW

File: internal/namespaces/rdb/v1/custom_instance_clone.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	"github.com/scaleway/scaleway-sdk-go/api/rdb/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func instanceCloneBuilder(c *core.Command) *core.Command {
+	c.WaitFunc = func(ctx context.Context, argsI, respI interface{}) (interface{}, error) {
+		api := rdb.NewAPI(core.ExtractClient(ctx))
+		return api.WaitForInstance(&rdb.WaitForInstanceRequest{
+			InstanceID:    respI.(*rdb.Instance).ID,
+			Region:        argsI.(*rdb.CloneInstanceRequest).Region,
```
Comment (jerome-quere) [inline internal/namespaces/rdb/v1/custom_instance_clone.go:16]: Use region from response too

File: internal/namespaces/rdb/v1/custom_instance_create.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	"github.com/scaleway/scaleway-sdk-go/api/rdb/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func instanceCreateBuilder(c *core.Command) *core.Command {
+	c.WaitFunc = func(ctx context.Context, argsI, respI interface{}) (interface{}, error) {
+		api := rdb.NewAPI(core.ExtractClient(ctx))
+		return api.WaitForInstance(&rdb.WaitForInstanceRequest{
+			InstanceID:    respI.(*rdb.Instance).ID,
+			Region:        argsI.(*rdb.CreateInstanceRequest).Region,
```
Comment (jerome-quere) [inline internal/namespaces/rdb/v1/custom_instance_create.go:16]: Use region from response

File: internal/namespaces/rdb/v1/custom_instance_upgrade.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	"github.com/scaleway/scaleway-sdk-go/api/rdb/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func instanceUpgradeBuilder(c *core.Command) *core.Command {
+	c.WaitFunc = func(ctx context.Context, argsI, respI interface{}) (interface{}, error) {
+		api := rdb.NewAPI(core.ExtractClient(ctx))
+		return api.WaitForInstance(&rdb.WaitForInstanceRequest{
+			InstanceID:    respI.(*rdb.Instance).ID,
+			Region:        argsI.(*rdb.UpgradeInstanceRequest).Region,
```
Comment (jerome-quere) [inline internal/namespaces/rdb/v1/custom_instance_upgrade.go:16]: Use region from response

File: internal/namespaces/rdb/v1/custom_instance.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+const (
+	instanceActionTimeout = 20 * time.Minute
+)
+
+type serverWaitRequest struct {
+	InstanceID string
+	Region     scw.Region
+}
+
+func instanceWaitCommand() *core.Command {
```
Comment (jerome-quere) [inline internal/namespaces/rdb/v1/custom_instance.go:22]: add a test for hand written command

Comment (remyleone) [reply internal/namespaces/rdb/v1/custom_instance.go:22]: Fixed

**Pull request merged**